packages/ra-ui-materialui/src/list/pagination/Pagination.tsx

```diff
@@ -94,7 +94,7 @@ export const Pagination: FC<PaginationProps> = memo(props => {
     }
 
     // Avoid rendering TablePagination if "page" value is invalid
-    if (total === 0 || page < 1 || (total != null && page > totalPages)) {
+    if (total === 0 || page < 1 || (total != null && page > totalPages!)) {
```

Comment:
**question**: `totalPages` could be `undefined` isn't it?

Reply:
not if total is not null